File: cameleon-backend/src/main/java/at/htl/entity/OverlayObjectText.java
package at.htl.entity;

import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "overlay_object_text")
public class OverlayObjectText extends OverlayObject {

    private String text;
    private String color;

    public OverlayObjectText() {
    }

    public OverlayObjectText(Cam cam, double x, double y, double scale, double opacity, String text, String color) {
        super(cam, x, y, scale, opacity);
        this.text = text;
        this.color = color;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }
}

File: cameleon-frontend/src/app/models/overlay-object-text.ts
export class OverlayObjectText {
  id: number;
  cam: number;
  x: number;
  y: number;
  scale: number;
  opacity: number;
  text: string;
  color: string;

  constructor(id: number, cam: number, x: number, y: number, scale: number, opacity: number, text: string, color: string) {
    this.id = id;
    this.cam = cam;
    this.x = x;
    this.y = y;
    this.scale = scale;
    this.opacity = opacity;
    this.text = text;
    this.color = color;
  }
}
